### PR TITLE
[CELEBORN-391][Flink] Refine register/release synchronization

### PR DESCRIPTION
**Pull request by RexXiong**
### What changes were proposed in this pull request?
1.Refine register/release synchronization
2.initial credit may equal to 0, we can bypass that.

### Why are the changes needed?
Avoid useless synchronization.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
UT & Tpcds 1T pass

### CODE REVIEW

**Comment by RexXiong** [comment]
@waitinfuture @FMX

File: common/src/main/java/org/apache/celeborn/common/network/server/BufferStreamManager.java [hunk 1]
```diff
@@ -111,37 +111,43 @@ public long registerStream(
     long streamId = nextStreamId.getAndIncrement();
     streams.put(streamId, new StreamState(channel, fileInfo.getBufferSize()));
     logger.debug("Register stream start streamId: {}, fileInfo: {}", streamId, fileInfo);
+    MapDataPartition mapDataPartition;
     synchronized (activeMapPartitions) {
-      MapDataPartition mapDataPartition = activeMapPartitions.get(fileInfo);
+      mapDataPartition = activeMapPartitions.get(fileInfo);
```
[the remark this answers is not in the document]
Comment (RexXiong) [reply common/src/main/java/org/apache/celeborn/common/network/server/BufferStreamManager.java:116]: Not really, actually, Map data partition may closed and removed out by recycle thread.

File: common/src/main/java/org/apache/celeborn/common/network/server/BufferStreamManager.java [hunk 2]
```diff
@@ -111,37 +111,43 @@ public long registerStream(
...
       if (mapDataPartition == null) {
         mapDataPartition = new MapDataPartition(fileInfo);
         activeMapPartitions.put(fileInfo, mapDataPartition);
       }
       mapDataPartition.addStream(streamId);
```
[the remark this answers is not in the document]
Comment (RexXiong) [reply common/src/main/java/org/apache/celeborn/common/network/server/BufferStreamManager.java:121]: This guarantee recycle thread cannot recycle this map data partition.

**Review by waitinfuture — APPROVED** [review]
LGTM, thanks!

**Pull request merged**